### PR TITLE
Install `nginx-prometheus-exporter`  and export metrics over `cos`

### PR DESCRIPTION
**Pull request by addyess**
* will first require https://github.com/charmed-kubernetes/jenkins/pull/1416 to build charms

Expose nginx metrics (which implements the load-balancer) over the cos relation by mapping the nginx status stub through the `nginx-prometheus-exporter`

These changes add that exporter binary as a charm resource which is an arch specific binary.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -113,12 +125,13 @@ def _configure_nginx(self, servers: Dict[int, Set]):
         """
         self.nginx.configure_site(
             "apilb",
-            Path(Path.cwd() / "templates/apilb.conf"),
+            TEMPLATES_PATH / "apilb.conf",
```
Comment (mateoflorido) [inline src/charm.py:128]: Ugh, nice catch 🎣

**Review by mateoflorido — CHANGES_REQUESTED** [review]
LGTM. Just one last thing, we have to change the `nginx_rule.yaml` file to check the `nginx_up` metric instead of the systemd service status.

**Comment by addyess** [comment]
> LGTM. Just one last thing, we have to change the `nginx_rule.yaml` file to check the `nginx_up` metric instead of the systemd service status.

I'm going to relate today to cos and see how things look in prometheus/grafana etc...

File: tests/unit/test_charm.py [hunk 2]
```diff
@@ -46,27 +46,24 @@ def test__configure_hacluster(self):
...
-            self.mock_nginx.return_value.remove_default_site.assert_called_once()
-            mock_write.assert_called_once()
+    def test_configure_nginx_sites(self):
+        servers = {80: {("backend1", 8080), ("backend2", 8081)}}
+        self.harness.update_config({"proxy_read_timeout": 10})
+        self.charm._configure_nginx_sites(servers)
+        assert self.mock_nginx.return_value.configure_site.mock_calls == [
+            call(
+                "apilb",
+                Path.cwd() / "templates" / "apilb.conf",
```
Comment (mateoflorido) [inline tests/unit/test_charm.py:56]:
```suggestion
                TEMPLATES_PATH / "apilb.conf",
```

**Review by mateoflorido — APPROVED** [review]
LGTM!

**Pull request merged**